src/screens/Inventory/AddInventory.tsx

```diff
@@ -61,29 +94,66 @@ function AddInventory (props: AddInventoryProps) {
     // Navigate to new inventory item's profile page
     setCurrentInventoryIdInRedux(inventory.id);
     history.replace(`item`);
-  }
-  const handleSelectItem = (event: React.ChangeEvent<{ }>, value: string | null) => {
-      setSelectedProductId(value || "");
-  }
+  };
+  
+  const handleSelectItem = (event: React.ChangeEvent<{}>, value: string | null) => {
+    setSelectedProductId(value || '');
+  };
 
   const handleStartingAmountInput = (event: React.ChangeEvent<{ value: unknown }>) => {
     setStartingAmount(parseFloat(event.target.value as string) || 0);
-  }
+  };
+
+  const handleNewProductName = (event: React.ChangeEvent<{ value: unknown }>) => {
+    setNewProductName((event.target.value as string) || '');
+  };
+
+  const handleUnitInput = (event: React.ChangeEvent<{ value: unknown }>) => {
+    setUnit((event.target.value as string) || '');
+  };
+
+  const filter = createFilterOptions<string>();
 
   return (
     <BaseScreen title="New Inventory" leftIcon="backNav">
       <form noValidate className={classes.content} onSubmit={() => false}>
         <FormControl variant="outlined" className={classes.formControl}>
           <Autocomplete
-            id = "select-item"
-            options = {productOptions}
-            style = {{ marginBottom: 24 }}
-            getOptionLabel={(productId) => `${products[productId]?.name} (${products[productId]?.unit})`}
-            renderInput={(params) => <MaterialTextField {...params} label="Item" variant="outlined"/>}
+            value={selectedProductId}
+            style={{ marginBottom: 20 }}
             onChange={handleSelectItem}
+            filterOptions={(options, params) => {
+              const filtered = filter(options, params);
+              filtered.push(NEW_PRODUCT_LABEL);
+              return filtered;
+            }}
```

Comment:
Based on https://material-ui.com/components/autocomplete/#createfilteroptions-config-filteroptions and the API documentation, I don't think I understand what's going on in this line. I thought you're just pass the `filter` you create to `filterOptions`. It might be the reason it's not working for me?

![image](https://user-images.githubusercontent.com/35501399/113425536-e3b32000-9386-11eb-9eaa-a0b305492f31.png)

Reply:
I just realized what this piece of code is trying to do (always show the `NEW_PRODUCT_LABEL`). Still, It's not working for me for some reason :(

Reply:
Ah I was going off of the code demo in this section https://material-ui.com/components/autocomplete/#creatable because I originally had it in the same format (where you type in the name of the new thing you want to create and it shows up as a new option like `Add "what you typed"`).

Reply:
Makes sense, thanks!